### PR TITLE
2.0 update of Limiting identity by scheme doc

### PR DESCRIPTION
**Pull request by scottaddie**
Fixes https://github.com/aspnet/Docs/issues/4510

[Internal Review Page](https://review.docs.microsoft.com/aspnet/core/security/authorization/limitingidentitybyscheme?branch=pr-en-us-4531)

### CODE REVIEW

File: aspnetcore/security/authorization/limitingidentitybyscheme.md [hunk 1]
```diff
@@ -14,61 +14,115 @@ uid: security/authorization/limitingidentitybyscheme
...
-    AutomaticAuthenticate = false
-});
+    // Code omitted for brevity
+
+    services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
+        .AddCookie(options => {
+            options.LoginPath = "/Account/Unauthorized/";
+            options.AccessDeniedPath = "/Account/Forbidden/";
+        })
+        .AddJwtBearer();
```
Comment (Tratcher) [inline aspnetcore/security/authorization/limitingidentitybyscheme.md]: Missing parameters like Authority and Audience.

File: aspnetcore/security/authorization/limitingidentitybyscheme.md [hunk 2]
```diff
@@ -14,61 +14,115 @@ uid: security/authorization/limitingidentitybyscheme
...
+```
+
+In the preceding code, two authentication services have been added: one for cookies and one for bearer.
+
+>[!NOTE]
+>When adding multiple authentication middlewares, ensure that no middleware is configured to run automatically. You do this by supplying an argument, such as `CookieAuthenticationDefaults.AuthenticationScheme`, to the `AddAuthentication` method. If you fail to do this, filtering by scheme will not work.
```
Comment (Tratcher) [inline aspnetcore/security/authorization/limitingidentitybyscheme.md]: ?? `services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)` makes cookies run automatically. You can override this per endpoint as shown by the Authorize attributes below.

File: aspnetcore/security/authorization/limitingidentitybyscheme.md [hunk 3]
```diff
@@ -14,61 +14,115 @@ uid: security/authorization/limitingidentitybyscheme
...
+>When adding multiple authentication middlewares, ensure that no middleware is configured to run automatically. You do this by setting the `AuthenticationOptions.AutomaticAuthenticate` property to false. If you fail to do this, filtering by scheme will not work.
+
+---
 
 ## Selecting the scheme with the Authorize attribute
 
-As no authentication middleware is configured to automatically run and create an identity you must, at the point of authorization choose which middleware will be used. The simplest way to select the middleware you wish to authorize with is to use the `ActiveAuthenticationSchemes` property. This property accepts a comma delimited list of Authentication Schemes to use. For example;
+No authentication middleware is configured to automatically run and create an identity. At the point of authorization, you choose which middleware will be used. The simplest way to select the middleware with which you wish to authorize is to use the `ActiveAuthenticationSchemes` property. This property accepts a comma-delimited list of authentication schemes to use. For example:
```
Comment (Tratcher) [inline aspnetcore/security/authorization/limitingidentitybyscheme.md]: Authorize specify the auth scheme to use regardless of if there is a default.

**Comment by scottaddie** [comment]
@Tratcher Please take another look.

File: aspnetcore/security/authentication/cookie.md [hunk 4]
```diff
@@ -71,7 +71,7 @@ The code snippets above configure some or all of the following options:
 
 * `AccessDeniedPath` - This is the relative path to which requests redirect when a user attempts to access a resource but does not pass any [authorization policies](xref:security/authorization/policies#security-authorization-policies-based) for that resource.
 
-* `AuthenticationScheme` - This is a value by which a particular cookie authentication scheme is known. This is useful when there are multiple instances of cookie authentication and you want to [limit authorization to one instance](xref:security/authorization/limitingidentitybyscheme#security-authorization-limiting-by-scheme).
+* `AuthenticationScheme` - This is a value by which a particular cookie authentication scheme is known. This is useful when there are multiple instances of cookie authentication and you want to [limit authorization to one instance](xref:security/authorization/limitingidentitybyscheme).
```
Comment (Rick-Anderson) [inline aspnetcore/security/authentication/cookie.md]: * `AuthenticationScheme` - This is a value by which a particular cookie authentication scheme is known. This is useful when there are multiple instances of cookie authentication and the app needs to [limit authorization to one instance](xref:security/authorization/limitingidentitybyscheme).

**Comment by HaoK** [comment]
General comments: Most references to middleware should probably be replaced with something more like "handler"?

I think the Note: about AddAuthentication feels a bit off, as you mention in the [Authorize] section the default doesn't affect that, I think it would be better if that note just states that specifying the default results in context.User being set to that identity, if that behavior isn't desired, turn it off.  Its perfectly reasonable to set the default to cookie, and have some API actions specify on the "bearer" for example.

I think the title might be better as "How to authorize with a specific scheme" or something like that instead of "limiting identity by scheme" which doesn't seem all that clear what that means

File: aspnetcore/security/authorization/limitingidentitybyscheme.md [hunk 5]
```diff
@@ -14,61 +14,120 @@ uid: security/authorization/limitingidentitybyscheme
...
+        .AddJwtBearer(options => {
+            options.Audience = "http://localhost:5001/";
+            options.Authority = "http://localhost:5000/";
+        });
+```
+
+In the preceding code, two authentication services have been added: one for cookies and one for bearer.
+
+>[!NOTE]
+>When adding multiple authentication middlewares, ensure that no middleware is configured to run automatically. Invoking `AddAuthentication` with no arguments ensures that no middleware is configured to run automatically. If the app invokes `AddAuthentication` with arguments, filtering by scheme doesn't work. For example, calling `AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)` makes cookies run automatically.
```
Comment (Tratcher) [inline aspnetcore/security/authorization/limitingidentitybyscheme.md]: There's nothing wrong with having a default for this scenario. You can still override it by specifying a different scheme in the authorize attribute.

File: aspnetcore/security/authorization/limitingidentitybyscheme.md [hunk 6]
```diff
@@ -14,61 +14,120 @@ uid: security/authorization/limitingidentitybyscheme
...
+If you prefer to specify the desired schemes in [policy](xref:security/authorization/policies#security-authorization-policies-based), you can set the `AuthenticationSchemes` collection when adding your policy:
 
 ```csharp
-options.AddPolicy("Over18", policy =>
+services.AddAuthorization(options =>
 {
-    policy.AuthenticationSchemes.Add("Bearer");
-    policy.RequireAuthenticatedUser();
-    policy.Requirements.Add(new Over18Requirement());
+    options.AddPolicy("Over18", policy =>
```
Comment (Tratcher) [inline aspnetcore/security/authorization/limitingidentitybyscheme.md]: Also show using this policy in the Authorization attribute.

**Comment by Rick-Anderson** [comment]
@scottaddie  make sure you squash before merging

**Pull request merged**